src/Middleware/CORS/src/Infrastructure/DefaultCorsPolicyProvider.cs

```diff
@@ -30,7 +30,13 @@ public Task<CorsPolicy> GetPolicyAsync(HttpContext context, string policyName)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            return Task.FromResult(_options.GetPolicy(policyName ?? _options.DefaultPolicyName));
+            policyName ??= _options.DefaultPolicyName;
+            if (_options.PolicyMap.TryGetValue(policyName, out var result))
+            {
+                return result.policyTask;
+            }
+
+            return Task.FromResult<CorsPolicy>(null);
```

Comment:
Does this benefit from caching? Or does .NET do this for you?

Reply:
As far as I know there is no caching.

You should return a static readonly instance.